src/gtk/textctrl.cpp

```diff
@@ -124,6 +124,41 @@ static void wxGtkTextApplyTagsFromAttr(GtkWidget *text,
         }
     }
 
+    if( attr.HasFontUnderlinedWithEffect() )
+    {
+        GtkTextTag *tag1;
+        PangoUnderline pg_style;
+        if( !(attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_NONE || attr.GetUnderlinedType() == wxTEXT_ATTR_UNDERLINE_SOLID ) )
```

Comment:
Please use `switch` instead. This would immediately show that some of the checks inside this `if` test for impossible things.